### PR TITLE
Remove non-self-start initialization

### LINKED ISSUE #1040: Remove non-self starting time-stepping code
# Feature request:

With self-starting working for time steppers, there is no reason to support non-self starting time stepping.

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Desired feature:

- [ ] Detail 1
- [ ] Detail 2
- [ ] Detail 3

### PR DESCRIPTION
**Pull request by wthrowe**
Fixes #1040

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp [hunk 1]
```diff
@@ -321,47 +318,18 @@ struct InitializeElement {
...
-          hist_vars.assign_subset(solution.variables(
-              inertial_coords, past_t.value(), typename Vars::tags_list{}));
-          dt_hist_vars.assign_subset(solution.variables(
-              inertial_coords, past_t.value(), typename DtVars::tags_list{}));
-          history.insert_initial(past_t, std::move(hist_vars),
-                                 std::move(dt_hist_vars));
-        }
-      }
+      // This is stored as Next<TimeId> and will be used to update
+      // TimeId at the start of the algorithm.
```
Comment (fmahebert) [inline src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp]: Can you clarify (in a github comment for now) what this means? I don't see any `Next<TimeId>` here... what makes you say it is stored as a `Next<TimeId>`?

Comment (wthrowe) [reply src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp]: It's the DataBox tag.  That would probably be a lot clearer if I included the `Tags::` prefixes.

Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp]: Okay, I understand now. I agree that adding prefixes would help 👍

**Review by kidder — DISMISSED** [review]

**Review by fmahebert — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Comment by kidder** [comment]
travis failures are timeouts

**Pull request merged**